src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AuthorizationMessageHandler.cs

```diff
@@ -36,6 +38,13 @@ public class AuthorizationMessageHandler : DelegatingHandler
         {
             _provider = provider;
             _navigation = navigation;
+
+            // Invalidate the cached _lastToken when the authentication state changes
+            if (_provider is AuthenticationStateProvider authStateProvider)
+            {
+                _authenticationStateChangedHandler = _ => { _lastToken = null; };
```

Comment:
You could have done this without the field by creating a method and assigning the method group instead, for future reference. But I'm ok if you leave it as is.